app/controllers/homes_controller.rb

```diff
@@ -1,7 +1,18 @@
 class HomesController < ApplicationController
-  # feel free to replace this controller
+  before_action :authenticate_user!, except: [:index]
 
   def index
+  end
+
+  def authenticated
+    render :index
+  end
+
+  protected 
 
+  def authenticate_user!
```

Comment:
Because this method is defined as `protected`, your other controllers are able to access it! You can use the above `before action` in the monster controller for the create action as well to ensure that the endpoint can't be hit unless a user is signed in (even though you're also blocking access to the new monster page, this would be an extra precaution!)